storage/memory.go

```diff
@@ -352,7 +351,7 @@ func (s *MemoryStore) Authenticate(_ context.Context, name string, secret string
 		return fosite.ErrNotFound
 	}
 	if rel.Password != secret {
-		return errors.New("Invalid credentials")
+		return fosite.ErrNotFound
```

Comment:
This store is really just a reference. I think it’s user friendly to retutn an understandable error message here. Not sure if there is value in making this into a generic „not found“ message

Reply:
To provide some context, the code that is calling the storage i.e. (ResourceOwnerPasswordCredentialsGrantHandler) (excerpt pasted below) expects only ErrNotFound from implementations of storage and injects its own top-level message. Earlier the calling code was propagating the error as 500 server error. I am assuming changing the calling code to react to a more appropriate error constant will break the contract with other implementations of storage and that is not an option.

} else if err := c.ResourceOwnerPasswordCredentialsGrantStorage.Authenticate(ctx, username, password); errors.Is(err, fosite.**ErrNotFound**) {
		return errorsx.WithStack(fosite.ErrInvalidGrant.WithHint("Unable to authenticate the provided username and password credentials.").WithWrap(err).WithDebug(err.Error()))

Reply:
@aeneasr Please let me know if you have another way to solve this or you prefer to leave it the way it is. Thanks.

Reply:
Hey - totally my bad, I forgot to reply here! I think we have a "WithDebug" helper on fosite errors which we could use here to provide more context, so:

```suggestion
		return fosite.ErrNotFound.WithDebug("Invalid credentials")
```